packages/next/src/build/utils.ts

```diff
@@ -1563,9 +1563,9 @@ export async function isPageStatic({
           useCache: true,
           distDir,
         })
-        const mod = (
+        const mod =
           await runtime.context._ENTRIES[`middleware_${edgeInfo.name}`]
-        ).ComponentMod
+            .ComponentMod
```

Comment:
awaiting the `module` seems more correct to me...

This should be fixed in the edge runtime